cabbage/task_worker.py

```diff
@@ -0,0 +1,84 @@
+import functools
+import logging
+import select
+from typing import Any, Callable
+
+from cabbage import exceptions, postgres, tasks
+
+logger = logging.getLogger(__name__)
+
+
+SOCKET_TIMEOUT = 5  # seconds
+
+
+def worker(
+    task_manager: tasks.TaskManager, queue: str, timeout: int = SOCKET_TIMEOUT
+) -> None:
+    conn = postgres.get_global_connection()
+
+    with postgres.get_dict_cursor(conn) as curs:
+        postgres.listen_queue(curs, queue)
+        infinite_loop(
+            functools.partial(
+                one_loop,
+                task_manager=task_manager,
+                queue=queue,
+                curs=curs,
+                timeout=timeout,
+            )
+        )
+
+
+def infinite_loop(func: Callable) -> None:
+    while True:
+        func()
+
+
+def one_loop(task_manager: tasks.TaskManager, queue: str, curs: Any, timeout) -> None:
+    process_tasks(task_manager=task_manager, queue=queue, curs=curs)
+    logger.debug("waiting")
+    select.select([curs.connection], [], [], timeout)
+
+
+def process_tasks(task_manager: tasks.TaskManager, queue: str, curs: Any) -> None:
+
+    for task_row in postgres.get_tasks(cursor=curs, queue=queue):  # pragma: no branch
+        if task_row["id"] is None:
+            break
+
+        assert isinstance(task_row["id"], int)
+        task_id = task_row["id"]
+
+        status = "error"
+        try:
+            logger.debug(f"""About to run task from row {task_row})""")
+            call_task(task_manager=task_manager, task_row=task_row)
+            status = "done"
+        except exceptions.TaskError:
+            pass
+        finally:
+            logger.debug(f"Calling finish_task({task_id}, {status})")
+            postgres.finish_task(cursor=curs, task_id=task_id, status=status)
+
+
+def call_task(task_manager: tasks.TaskManager, task_row: dict) -> None:
+    task_name = task_row["task_type"]
+    try:
+        task = task_manager.tasks[task_name]
+    except KeyError:
+        raise exceptions.TaskNotFound(task_row)
+
+    pk = task_row["id"]
+
+    task_run = tasks.TaskRun(task=task, id=pk, lock=task_row["targeted_object"])
+    kwargs = task_row["args"]
+
+    description = f"{task.queue}.{task.name}.{pk}({kwargs})"
```

Comment:
could we use a generic message and the variables in `extra=` ?

Reply:
We'll do that in a dedicated PR (issue #4)